server/game/cards/02_SHD/units/GentleGiant.ts

```diff
@@ -0,0 +1,29 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { WildcardCardType } from '../../../core/Constants';
+
+export default class GentleGiant extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8190373087',
+            internalName: 'gentle-giant'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addOnAttackAbility({
+            title: 'You may heal damage from another unit equal to the damage on this unit',
+            optional: true,
+            targetResolver: {
+                cardCondition: (card, context) => card !== context.source,
+                cardTypeFilter: WildcardCardType.Unit,
+                immediateEffect: AbilityHelper.immediateEffects.heal((context) => ({
```

Comment:
nit: just for readability, could put this all in one since there's just one property. if you like it spread out though feel free to keep it that way